File: 242. Valid Anagram/242. Valid Anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length !== t.length) {
    return false;
  }

  const sCounter = new Array(26).fill(0);
  const tCounter = new Array(26).fill(0);

  for (let i = 0; i < s.length; i++) {
    sCounter[s.charCodeAt(i) - "a".charCodeAt(0)]++;
    tCounter[t.charCodeAt(i) - "a".charCodeAt(0)]++;
  }

  for (let i = 0; i < 26; i++) {
    if (sCounter[i] !== tCounter[i]) {
      return false;
    }
  }

  return true;
};

File: 242. Valid Anagram/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_counter = [0] * 26
        t_counter = [0] * 26

        for i in range(len(s)):
            s_counter[ord(s[i]) - ord("a")] += 1
            t_counter[ord(t[i]) - ord("a")] += 1

        return s_counter == t_counter
